src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -447,6 +447,84 @@ JRT_LEAF(jdouble, SharedRuntime::l2d(jlong x))
   return (jdouble)x;
 JRT_END
 
+JRT_LEAF(jshort, SharedRuntime::f2hf(jfloat  x))
```

Comment:
Is it necessary/desirable to have a transliteration of the Java implementation here rather than just using the Java library implementation?

Reply:
@jddarcy Thanks for your comment. I am not sure if there is a way of using Java library implementation here.

Reply:
I was under the impression that if a platform didn't have special support for the functionality in question it could not have the intrinsic and just use the Java library implementation?

Reply:
@jddarcy This code is for constant expression evaluation during C2 compilation.